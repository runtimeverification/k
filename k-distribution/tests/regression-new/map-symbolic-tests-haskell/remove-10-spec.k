// Copyright (c) Runtime Verification, Inc. All Rights Reserved.
// generated by gen-tests.sh

requires "map-tests.k"

module REMOVE-10-SPEC
    imports MAP-TESTS

    claim <k> remove ( (MAP:Map X:MyId |-> 1) [Y:MyId <- undef] [ Z:MyId ] ) => . </k> requires Z  ==K X

endmodule
