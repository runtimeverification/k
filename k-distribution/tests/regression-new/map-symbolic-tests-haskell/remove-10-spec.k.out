  <k>
    removeResult ( ( MAP
    Z:MyId |-> 1 ) [ Y:MyId <- undef ] [ Z:MyId ] ~> .K ) ~> .K
  </k>
#And
  {
    false
  #Equals
    Z:MyId in_keys ( MAP )
  }
#And
  {
    true
  #Equals
    Z:MyId in_keys ( ( MAP
    Z:MyId |-> 1 ) [ Y:MyId <- undef ] )
  }
[Error] Prover: backend terminated because the configuration cannot be rewritten further. See output for more details.
